File: CS/E1293/S131267/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Views.Base;

namespace S131267 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            dataSet1.ReadXml(@"..\..\nwind.xml");
            bindingSource1.DataMember = "Categories";
        }

        private void OnGridViewDoubleClick(object sender, EventArgs e) {
            GridView gView = (GridView)sender;
            GridHitInfo hInfo = gView.CalcHitInfo(gView.GridControl.PointToClient(Control.MousePosition));
            if (hInfo.InRowCell) {
                new EditForm(Control.MousePosition, gView.Columns, bindingSource1, BindingContext).ShowDialog(this);
                gView.UpdateCurrentRow();
            }
        }
    }
}

File: VB/E1293/S131267/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Views.Base

Namespace S131267
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			dataSet1.ReadXml("..\..\nwind.xml")
			bindingSource1.DataMember = "Categories"
		End Sub

		Private Sub OnGridViewDoubleClick(ByVal sender As Object, ByVal e As EventArgs) Handles gridView1.DoubleClick
			Dim gView As GridView = CType(sender, GridView)
			Dim hInfo As GridHitInfo = gView.CalcHitInfo(gView.GridControl.PointToClient(Control.MousePosition))
			If hInfo.InRowCell Then
				CType(New EditForm(Control.MousePosition, gView.Columns, bindingSource1, BindingContext), EditForm).ShowDialog(Me)
				gView.UpdateCurrentRow()
			End If
		End Sub
	End Class
End Namespace